lib/dialyzer/test/typer_SUITE.erl

```diff
@@ -51,6 +55,39 @@ smoke(Config) ->
           ok
     end.
 
+otp_6296_no_spec_flag_does_not_break_records(Config) ->
+    Code = <<"-module(otp_6296).
```

Comment:
One more thing. `otp_xxxx` is for our internally created tickets, not for github issues. To avoid confusion:

```suggestion
gh_6296_no_spec_flag_does_not_break_records(Config) ->
    Code = <<"-module(gh_6296).
```